secrets/alibaba/alibaba_ssm.go

```diff
@@ -0,0 +1,222 @@
+package alibabassm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/secrets"
+	openapi "github.com/alibabacloud-go/darabonba-openapi/v2/client"
+	oos20190601 "github.com/alibabacloud-go/oos-20190601/v4/client"
+	util "github.com/alibabacloud-go/tea-utils/v2/service"
+	"github.com/alibabacloud-go/tea/tea"
+	"github.com/hashicorp/go-hclog"
+)
+
+type AlibabaSsmManager struct {
+	// Local logger object
+	logger hclog.Logger
+
+	// The Alibaba region
+	region string
+
+	// Custom Alibaba endpoint, e.g. localstack
+	endpoint string
+
+	// The Alibaba SDK client
+	client *oos20190601.Client
+
+	// The base path to store the secrets in SSM Parameter Store
+	basePath string
+}
+
+func SecretsManagerFactory(
+	config *secrets.SecretsManagerConfig,
+	params *secrets.SecretsManagerParams) (secrets.SecretsManager, error) { //nolint
+
+	// Check if the node name is present
+	if config.Name == "" {
+		return nil, errors.New("no node name specified for Alibaba SSM secrets manager")
+	}
+
+	// Check if the extra map is present
+	if config.Extra == nil || config.Extra["region"] == nil || config.Extra["ssm-parameter-path"] == nil {
+		return nil, errors.New("required extra map containing 'region' and 'ssm-parameter-path' not found for aws-ssm")
```

Comment:
should we call this OOS instead of SSM?

Reply:
I was thinking how to name it and one of the solutions was oos. However we don't use AWS SSM but parameter store instead of. So AWS SSM is not SSM also, hence I named it the same as alibaba-ssm because we know for what we are using it and SSM is phrase which is Blade established :)

Reply:
AWS's SSM means Systems Manager, and within Systems Manager, there is a Parameter Store. Likewise, Alibaba Cloud's OOS, or CloudOps Orchestration Service, also has a Parameter Store. I believe we called it wrong from the beginning, but I agree that we simply called it `ssm`.

Reply:
SSM doesn't look like shortcut from Systems Manager, it has an extra S.

Reply:
Just like OOS (which stands for CloudOps Orchestration Service), but it is what it is.

Reply:
References: [AWS SSM](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html) and [Alibaba Cloud OOS](https://www.alibabacloud.com/help/en/ros/use-cases/use-oos-parameter-store-to-create-parameters-for-ros-templates)